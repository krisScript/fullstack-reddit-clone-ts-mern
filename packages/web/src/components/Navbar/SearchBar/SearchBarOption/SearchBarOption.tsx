import React from 'react';
import AutoCompleteCommunity from '../../../../types/AutoCompleteCommunity';
import { Avatar } from 'antd';
import { Typography } from 'antd';
const { Text } = Typography;
interface SearchBarOptionProps {
  community: AutoCompleteCommunity;
}
const SearchBarOption: React.FC<SearchBarOptionProps> = ({ community }) => {
  return (
    <div>
      {' '}
      <Avatar
        size="small"
        src={`http://localhost:8080/images/${community.data.theme.icon}`}
        alt={`${community.data.name} icon`}
      />
      <Text style={{ padding: '10px' }}>{community.data.name}</Text>
      <Text style={{ padding: '10px' }}>
        {community.data.subscribers} members
      </Text>
    </div>
  );
};

export default SearchBarOption;
