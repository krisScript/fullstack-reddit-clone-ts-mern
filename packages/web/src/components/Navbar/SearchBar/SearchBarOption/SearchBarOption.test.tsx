import React from 'react';
import { render, fireEvent, waitForElement } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import SearchBarOption from './SearchBarOption';
describe('<SearchBarOption/>', () => {
  const community = {
    data: {
      name: 'testName',
      subscribers: 1000,
      score: 1,
      _id: 'testId',
      theme: {
        icon: 'test.svg',
        colors: {
          base: 'baseColor',
          highlight: 'highlightColor',
        },
      },
    },
    links: {
      self: 'testLink',
    },
  };
  const { container, getByText, getByAltText } = render(
    <SearchBarOption community={community} />,
    { wrapper: BrowserRouter },
  );
  const { name, subscribers } = community.data;
  const { icon } = community.data.theme;
  it('snapshot', () => {
    expect(container).toMatchSnapshot();
  });
  it(`should exist`, async () => {
    const element = getByText(name);
    expect(element).toBeTruthy();
  });
  it(`should exist`, async () => {
    const element = getByText(`${subscribers} members`);
    expect(element).toBeTruthy();
  });
  it('should exist', async () => {
    const element = getByAltText(`${name} icon`) as HTMLImageElement;
    expect(element).toBeTruthy();
    expect(element.src).toMatch(`http://localhost:8080/images/${icon}`);
  });
});
