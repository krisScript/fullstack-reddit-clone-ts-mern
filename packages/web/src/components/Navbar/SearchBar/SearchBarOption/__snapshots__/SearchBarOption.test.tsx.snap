// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBarOption/> snapshot 1`] = `
<div>
  <a
    href="/communities/testId"
  >
    <span
      class="ant-avatar ant-avatar-sm ant-avatar-circle ant-avatar-image"
    >
      <img
        alt="testName icon"
        src="http://localhost:8080/images/test.svg"
      />
    </span>
    <span
      class="ant-typography"
      style="padding: 10px;"
    >
      testName
    </span>
    <span
      class="ant-typography"
      style="padding: 10px;"
    >
      1000
       members
    </span>
  </a>
</div>
`;
